versions/3.0.4.md

```diff
@@ -121,16 +121,18 @@ In order to preserve the ability to round-trip between YAML and JSON formats, YA
 
 An OpenAPI Description (OAD) MAY be made up of a single document or be divided into multiple, connected parts at the discretion of the author. In the latter case, [Reference Object](#reference-object) and [Path Item Object](#path-item-object) `$ref` keywords, as well as the [Link Object](#link-object) `operationRef` keyword, are used.
 
-Any document consisting entirely of an [OpenAPI Object](#openapi-object) is known as a **complete OpenAPI document**.
-In a multi-document description, the document containing the OpenAPI Object where parsing begins for a specific API's description is known as that API's **entry OpenAPI document**, or simply **entry document**.
+Any document consisting entirely of an [OpenAPI Object](#openapi-object) is known as a **syntactically complete OpenAPI document**.
+An OpenAPI document that does _not_ reference any other documents is known as a **self-contained OpenAPI document**.
+A single-document description is therefore _both_ syntactically complete _and_ self-contained.
+In a multi-document description, the document containing the OpenAPI Object where parsing begins for a specific API's description is known as that API's **entry document**.
```

Comment:
Here's a new suggestion for this part of the text. Let me know what you think of this (GitHub won't let me do this as a suggestion so I'll just quote it here:

> An OpenAPI Description (OAD) MAY describe certain features of the API "by reference", using the [Reference Object](#reference-object) or [Path Item Object](#path-item-object) `$ref` keywords, as well as the [Link Object](#link-object) `operationRef` keyword. When any of these references are used, the meaning of the OAD is equivalent to replacing the reference with description element at the referenced location.
> 
> Any of these references may specify content in a document other than the document containing the reference. Structuring an OAD as a single JSON or YAML document or composed from multiple, connected documents is at the discretion of the author. 
> 
> In a multi-document OAD, the document containing the OpenAPI Object where parsing begins for a specific API's description is known as that OAD's **entry document**.

Reply:
I'm stumbling over the sentence

> When any of these references are used, the meaning of the OAD is equivalent to replacing the reference with description element at the referenced location.

The "replacing" is only true for the Reference Object.

For Path Item Objects `$ref` means "merge the referenced object into this object and deal with conflicts as you like".

Link Objects are instructions on how to parameterize the referenced operation, the referenced operation does not replace the Link Object.

If this sentence is removed, I'd be ok with the suggestion.

Reply:
Okay ... I didn't realize these were different. I'll remove that sentence.

Reply:
@mikekistler I'm really reluctant to get into this much rewriting.  I was just trying to fix a tiny little thing on what I thought was already approved text.  We're on the verge of releasing this, is it truly necessary to revisit this in depth?

I don't agree with these further proposed changes and really do not want to delve more into referencing, or what "by reference" means, or whether references are replacements or not, etc.  in this section at all.

Reply:
Then perhaps we can just drop this PR altogether? Because honestly I think it only makes a muddy topic even muddier. 

But maybe it's just me ... and I suppose I should have spoken up when the earlier language was added. So let me see if I can just clear my review and let others decide.

Reply:
@mikekistler as I explained in another reply further down, this was a tiny fix to a very specific problem:  the misunderstanding that someone who is very familiar with the spec had when they read "complete document" to mean "self-contained with no external references."

I don't understand how my insertion of the word "syntactically" and defintion of a "self-contained" document makes anythig more confusing.  The things you brought up are pretty much all outside of the scope of that tiny fix, so I'm not sure why you are insisting on either a lot more change or no change at all.

If you want to put up a PR with more changes, I have no objection to that, I just object to expanding the scope of my very specific fix for a concrete, observed problem that could really cause a lot of confusion, with a more general debate on the wording.